yahtzee/mbramson/yahtzee/lib/yahtzee.ex

```diff
@@ -13,29 +13,31 @@ defmodule Yahtzee do
   def sixes(rolls), do: upper(rolls, 6)
 
   def pairs(rolls) do
-    rolls = Enum.sort(rolls)
-    rolls = Enum.reverse(rolls)
-    rolls_unique = Enum.uniq(rolls)
-    Enum.reduce_while(rolls_unique, 0,
-      fn x, acc ->
-        if match_exists(rolls, x, 2),
-        do: {:halt, acc + match_sum(rolls, x, 2)},
-        else: {:cont, acc}
-      end
-    )
-
+    rolls
+      |> Enum.group_by(fn(x) -> x end)
```

Comment:
Love the Enum.group_by function for this problem, but I'm not sure that it results in the most performant solution...